src/app-layout/visual-refresh-toolbar/notifications/index.tsx

```diff
@@ -23,8 +24,9 @@ export function AppLayoutNotificationsImplementation({
   children,
 }: AppLayoutNotificationsImplementationProps) {
   const { ariaLabels, stickyNotifications, setNotificationsHeight, verticalOffsets } = appLayoutInternals;
-  const ref = useRef<HTMLElement>(null);
-  useResizeObserver(ref, entry => setNotificationsHeight(entry.borderBoxHeight));
+  const [hasNotificationsContent, contentRef] = useContainerQuery(rect => rect.borderBoxHeight > 0);
```

Comment:
an extra resize observer may seem like a waste, but it allows us to resolve this rendering logic without talking outside this component, which is useful in the widget environment (compatibility with existing code)